server/service/validator.js

```diff
@@ -0,0 +1,66 @@
+const {BadRequestError} = require('../errorTypes/errorTypes')
+module.exports = {
+    validateAttributes: (modelAttributes, attributes) => {
+        //filtering attributes which aren't in the schema
+
+        const modelKeys = Object.keys(modelAttributes);
+        const filteredAttributes = Object.keys(attributes)
+            .filter(key => modelKeys.includes(key))
+            .reduce((acc, key) => Object.assign(acc, { [key]: attributes[key] }), {});
+
+
+        //check required
+        modelKeys.forEach(attributeKey => {
+            if (modelAttributes[attributeKey].required && filteredAttributes[attributeKey] == undefined) {
+                throw new BadRequestError(`${attributeKey} must be provided`)
+            }
+        });
+
+        //check types
+        Object.keys(filteredAttributes).forEach(key => {
+            const constructorName = filteredAttributes[key].constructor.name
+            if (constructorName.toLowerCase() !== modelAttributes[key].type) {
+                throw new BadRequestError(`Type of ${key} doesnt match`);
+            }
+        })
+        //length, in
+        modelKeys.forEach(modelKey => {
+            const attribute = filteredAttributes[modelKey];
+            if (attribute) {
+                const rulesKeys = Object.keys(modelAttributes[modelKey])
+                rulesKeys.forEach(ruleKey => {
+                    const ruleValue = modelAttributes[modelKey][ruleKey];
+                    switch (ruleKey) {
```

Comment:
please mode this rule checker into separate function and use it here

Reply:
Can i make it a class then? Cause I can't use this .validateRules() in this module.exports